src/index.ts

```diff
@@ -189,8 +190,45 @@ export const Rune: RuneExport = {
     // Reset randomness to be deterministic across plays
     Rune.deterministicRandom = Rune._randomNumberGenerator(Rune.getChallengeNumber())
   },
+  _getQueryParams: () => {
+    return decodeURI(window.location.search)
+        .replace('?', '')
+        .split('&')
+        .map(param => param.split('='))
+        .reduce((values, [ key, value ]) => {
+          values[ key ] = value
+          return values
+        }, {} as {[key: string]: string})
+  }
 }
 
+//Safari ios throttles requestAnimationFrame when user has not interacted with the iframe at least once.
+//In case the games are not using clicks (for instance only swiping), ios will not treat these interactions
+//with the iframe as user interacting. As a workaround, in the browser we will start overlay with
+//click events disabled and display an invisible div inside the iframe above the canvas.
+//This way the users will click on the transparent div element the very first time. We will let our client
+//know about it with _INITIAL_OVERLAY_CLICK event and the transparent div will remove itself.
+//Afterwards the play/pause will be once again fully controlled by our client.
+;(() => {
+  const queryParams = Rune._getQueryParams()
+
+  if (!!queryParams.enableInitialOverlayInBrowser && queryParams.enableInitialOverlayInBrowser === '1') {
+    document.addEventListener('DOMContentLoaded', function () {
+      const div = document.createElement('div')
+      div.setAttribute('style', "top: 0; bottom: 0; left: 0; right: 0; position: absolute; z-index: 9999;")
+
+      div.addEventListener('click', () => {
+        div.remove()
+        if (globalThis.postRuneEvent) {
+          globalThis.postRuneEvent({type: "_INITIAL_OVERLAY_CLICK"})
+        }
+      })
+      document.body.appendChild(div)
+    })
+  }
```

Comment:
> > @Gongreg this looks good to me! Had a question though — what happens in the race condition where the user presses the play button before the iFrame has fully loaded?
> 
> In that case nothing happens (we display play button as inactive). The iframe should load pretty fast (the initial dom structure without any assets/js). If users clicks play at that moment (while game is not initialized), we will go to "waitingToPlayState". Then as soon as the game fully loads (init called), we will start playing.

You mentioned that `we display play button as inactive`. I think that if the play button is disabled user should press on it again after the iFrame loads and the button gets enabled. I am just wondering will it be confusing if I press an inactive button and then after some time the game will just start itself.

Also, in this scenario, iFrame's div will not receive any clicks, so it will not remove the div until next user's click. I.e. user will need to click again anyway to remove the div. And if the game does not require clicks (just swipes) that may seem as broken.

Alternatively, iFrame could remove the div e.g. after game starts, however, I am not sure the first click (while iFrame is not loaded yet) would be considered by the browser as user interaction.

Hence, I would vote for enabling Play button only after receiving `INIT` from iFrame.

Reply:
I did it this way to replicate the way our app behaves. 
The next improvement will be to have a loading animation (like in the app).
 After users clicks on the iframe (which is identical to clicking play button in the app), we will hide the play button (only leave the loading animation) in our client to indicate that the user is waiting for the load to finish.

Reply:
And I guess we should not display the button as inactive while the games is loading, true.
Will create an issue for it.